components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -542,6 +542,23 @@ void BraveWalletProviderImpl::OnRequestEthereumPermissions(
     const std::vector<std::string>& accounts,
     mojom::ProviderError error,
     const std::string& error_message) {
+  // If the call was successful but the keyring is locked, then request an
+  // unlock.  After the unlock happens a new request will be made.
+  if (error == mojom::ProviderError::kSuccess &&
+      keyring_controller_->IsLocked()) {
+    if (pending_unlock_callback_) {
+      std::move(callback).Run(
+          std::vector<std::string>(),
+          mojom::ProviderError::kUserRejectedRequest,
+          l10n_util::GetStringUTF8(IDS_WALLET_ALREADY_IN_PROGRESS_ERROR));
+      return;
+    }
+    pending_unlock_callback_ = std::move(callback);
+    keyring_controller_->RequestUnlock();
+    delegate_->ShowPanel();
+    return;
+  }
```

Comment:
Why do we need to run `RequestEthereumPermissions` once and handle the unlock logic here? Can this block be moved to `RequestEthereumPermissions`?

Reply:
I had that originally but there was some test failures that way. For example when no wallet is setup we want to open the onboarding.  Basically we only want to do this request unlock logic when things would be successful but the keyring is locked.

Reply:
I see, thanks for the explanation.